### PR TITLE
Bugfix for hip08

### PR DESCRIPTION
**Pull request by Li-Weihang**
Various fixes on library of hip08.

### CODE REVIEW

**Comment by rleon** [comment]
Looks ok to me.

**Comment by Li-Weihang** [comment]
Changlog
v1->v2: Modify patch(1/7) to fix comments from Zengtao.

Please refer to:
https://patchwork.kernel.org/patch/11255123/

**Comment by Li-Weihang** [comment]
v2->v3:  Modify "ilog32(n) - 1" to "ilog32(n - 1)" in patch(1/7)

**Comment by Li-Weihang** [comment]
> Looks ok to me.

Hi Leon,
Could these fixes be merged into master? 
Thank you :)

File: providers/hns/hns_roce_u_hw_v2.c [hunk 1]
```diff
@@ -1040,7 +1039,9 @@ static int hns_roce_u_v2_post_recv(struct ibv_qp *ibvqp, struct ibv_recv_wr *wr,
 static void __hns_roce_v2_cq_clean(struct hns_roce_cq *cq, uint32_t qpn,
 				   struct hns_roce_srq *srq)
 {
+	int wqe_index;
```
Comment (jgunthorpe) [inline providers/hns/hns_roce_u_hw_v2.c]: is it really signed?

Comment (Li-Weihang) [reply providers/hns/hns_roce_u_hw_v2.c]: Thank you.
wqe_index should be uint16_t, I have modified it and re-pushed.

**Comment by rleon** [comment]
@jgunthorpe ???

**Pull request merged**